src/Stratis.Bitcoin.Features.SmartContracts/Wallet/SmartContractTransactionService.cs

```diff
@@ -144,7 +146,31 @@ public BuildCreateContractTransactionResponse BuildCreateTx(BuildCreateContractT
 
             ulong totalFee = (request.GasPrice * request.GasLimit) + Money.Parse(request.FeeAmount);
             var walletAccountReference = new WalletAccountReference(request.WalletName, request.AccountName);
-            var recipient = new Recipient { Amount = request.Amount ?? "0", ScriptPubKey = new Script(this.callDataSerializer.Serialize(txData)) };
+
+            byte[] serializedTxData = this.callDataSerializer.Serialize(txData);
+
+            Result<ContractTxData> deserialized = this.callDataSerializer.Deserialize(serializedTxData);
+
+            // We also want to ensure we're sending valid data: AKA it can be deserialized.
+            if (deserialized.IsFailure)
+            {
+                return BuildCreateContractTransactionResponse.Failed("Invalid data. If network requires code signing, check the code contains a signature.");
+            }
+
+            // HACK
+            // If requiring a signature, also check the signature.
+            if (this.network is ISignedCodePubKeyHolder holder)
```

Comment:
'Violates the architecture' ?  What does this mean? What architecture is being violated that we are so attached to? 

This approach is way more flexible than having a ConsensusOptions for every single new option - that's really bad.

Reply:
Also doesn't touch full node code at all.

Reply:
I think it was in general the approach we decided to take- put setting like that in consensus options instead of having them in several places around the codebase. 
@dangershony wdyt?

Reply:
@dangershony can you weigh in here whether it's blocking to use an interface to access things on the Network like this? Thanks